src/compiler/crystal/compiler.cr

```diff
@@ -295,7 +297,15 @@ module Crystal
 
       Crystal.timing("Codegen (linking)", @stats) do
         Dir.cd(output_dir) do
-          system %(#{CC} -o "#{output_filename}" "${@}" #{@link_flags} #{lib_flags}), object_names
+          if thin_lto
+            lto_cache_dir = "#{output_dir}/lto.cache"
+            Dir.mkdir_p(lto_cache_dir)
+            cc = ENV["CC"]? || "clang -flto=thin -Wl,-mllvm,-threads=#{n_threads},-cache_path_lto,#{lto_cache_dir},#{@release ? "-mllvm,-O2" : "-mllvm,-O0"}"
```

Comment:
Yeah, but we can specify link flags with `--link-flags`, we already force flags and libraries on `CC` anyway, and if we're going this way, there is `--cross-compile` to only generate objects.

 The thing is that Debian and Freebsd systems (and possibly others) don't provide a generic `clang` binary or link, only `clang-3.9`, `clang-4.0` and so on, so I can't use thin lto as the following call, or when I simply export `CC` in my `~/.profile`, which would result in non optimized binaries (many modules, no link-time optimizations):

```
$ CC=clang-4.0 shards build --release --lto=thin
```